RecoParticleFlow/PFClusterProducer/interface/PFRecHitQTests.h

```diff
@@ -539,10 +566,83 @@ class PFRecHitQTestHCALCalib29 : public PFRecHitQTestBase {
 	  
     }
 
+    bool test(reco::PFRecHit& hit,const HGCRecHit& rh,bool& clean){
+      return true;
+    }
+
  protected:
     float calibFactor_;
 };
 
+class PFRecHitQTestThresholdInMIPs : public PFRecHitQTestBase {
+ public:
+  PFRecHitQTestThresholdInMIPs() {
+
+  }
 
+  PFRecHitQTestThresholdInMIPs(const edm::ParameterSet& iConfig):
+    PFRecHitQTestBase(iConfig)
+    {
+      recHitEnergy_keV_ = iConfig.getParameter<bool>("recHitEnergyIs_keV");
+      threshold_ = iConfig.getParameter<double>("thresholdInMIPs");
+      mip_ = iConfig.getParameter<double>("mipValueInkeV");
+      recHitEnergyMultiplier_ = iConfig.getParameter<double>("recHitEnergyMultiplier");         
+    }
+
+    void beginEvent(const edm::Event& event,const edm::EventSetup& iSetup) {
+    }
+
+    bool test(reco::PFRecHit& hit,const EcalRecHit& rh,bool& clean) {
+      throw cms::Exception("WrongDetector")
+	<< "PFRecHitQTestThresholdInMIPs only works for HGCAL!";
+      return false;
+    }
+    bool test(reco::PFRecHit& hit,const HBHERecHit& rh,bool& clean) {
+      throw cms::Exception("WrongDetector")
+	<< "PFRecHitQTestThresholdInMIPs only works for HGCAL!";
+      return false;
+    }
+
+    bool test(reco::PFRecHit& hit,const HFRecHit& rh,bool& clean) {
+      throw cms::Exception("WrongDetector")
+	<< "PFRecHitQTestThresholdInMIPs only works for HGCAL!";
+      return false;
+    }
+    bool test(reco::PFRecHit& hit,const HORecHit& rh,bool& clean) {
+      throw cms::Exception("WrongDetector")
+	<< "PFRecHitQTestThresholdInMIPs only works for HGCAL!";
+      return false;
+    }
+
+    bool test(reco::PFRecHit& hit,const CaloTower& rh,bool& clean) {
+      throw cms::Exception("WrongDetector")
+	<< "PFRecHitQTestThresholdInMIPs only works for HGCAL!";
+      return false;
+    }
+
+    bool test(reco::PFRecHit& hit,const HGCRecHit& rh,bool& clean) {
+      const double newE = ( recHitEnergy_keV_ ? 
+			    1.0e-6*rh.energy()*recHitEnergyMultiplier_ :
+			    rh.energy()*recHitEnergyMultiplier_ );
+      /*
```

Comment:
here and below, please remove the commented out code.
If it's needed, change to LogDebug or LogTrace